server/src/business-layer/services/StripeService.ts

```diff
@@ -452,6 +458,58 @@ export default class StripeService {
     }
   }
 
+  private async isLastSpotTaken(bookingSlotId: string): Promise<boolean> {
+    const bookingDataService = new BookingDataService()
+    const bookingSlotService = new BookingSlotService()
+
+    const bookingSlot =
+      await bookingSlotService.getBookingSlotById(bookingSlotId)
+
+    const bookings = await bookingDataService.getBookingsBySlotId(bookingSlotId)
+    const bookingCount = bookings.length
+
+    const availableSlots = bookingSlot.max_bookings - bookingCount
+    return availableSlots <= 0
+  }
+
+  private async expireOtherCheckoutSessions(
+    bookingSlotId: string
+  ): Promise<void> {
+    try {
+      // Fetch all checkout sessions for the specific booking slot id
+      const sessions = await stripe.checkout.sessions.list({
```

Comment:
Suggest using the relevant method in stripeservice and get all sessions from 24 hrs (max) ago